frontend/src/lib/WebsocketConnection.tsx

```diff
@@ -462,47 +462,50 @@ function doHealthPing(
 
   xhr.timeout = timeoutMs
 
-  xhr.onload = () => {
-    if (xhr.responseText === "ok") {
-      resolver.resolve(uriNumber)
-    } else if (xhr.status === /* NO RESPONSE */ 0) {
-      const uri = uriList[uriNumber]
-      if (uri.startsWith("//localhost:")) {
-        const scriptname =
-          SessionInfo.isSet() && SessionInfo.current.commandLine.length
-            ? SessionInfo.current.commandLine[0]
-            : "yourscript.py"
-
+  xhr.onreadystatechange = () => {
+    if (xhr.readyState === /* DONE */ 4) {
```

Comment:
Also, this inner function is quite big and has a ton if `if/else` branches inside of it. So it's quite hard to read. It would be great to clean it up a bit.

For example:

```javascript
  xhr.onreadystatechange = () => {                                               
    if (xhr.readyState !== /* DONE */ 4) {                                       
      return                                                                     
    } else if (xhr.responseText === "ok") {                                      
      resolver.resolve(uriNumber)                                                
    } else if (xhr.status === /* NO RESPONSE */ 0) {                             
      retryAndMaybeShowRerunInstructions(...)                                    
    } else if (xhr.status === 403) {                                                
      retry(CORS_ERROR_MESSAGE)                                                  
    } else {                                                                     
      retry(                                                                     
        `Connection failed with status ${xhr.status}, ` +                        
          `and response "${xhr.responseText}".`                                  
      )                                                                             
    }                                                                               
  }
```

Reply:
Done